Source/Softswitch/inc/softswitch.h

```diff
@@ -1,46 +1,52 @@
 #ifndef _SOFTSWITCH_H_
 #define _SOFTSWITCH_H_ 
 
-#include <cstring>
 #include "poets_msg.h"
 
-#define NUM_SUP_BUFS 4
 #define MAX_LOG_MSG_BUFS 4
 
-class handler_log_msg
-{
-    public:
-    handler_log_msg();
-    uint32_t      pack_fmtstr(const char* &);
-    void          send_msg();
-      
-#ifndef TRIVIAL_LOG_HANDLER
-    // argument expansion and buffer packing
-    template<typename... U> inline void pack_log_args(U&&... args) {uint32_t _a[] = {0, (prepack_log_arg(args),0)...};}
-    // argument extractor
-    template<typename P> inline void prepack_log_arg(P arg) {pack_log_arg<P>(sizeof(P),&arg);};
-    // does the actual work
-    template<typename V> void pack_log_arg(uint32_t, V*);
+#define P_DEFAULT_LOG_LEVEL     2
+
+#ifndef P_LOG_LEVEL
+#define P_LOG_LEVEL P_DEFAULT_LOG_LEVEL
 #endif
-     
-    P_Sup_Msg_t*  log_msg[MAX_LOG_MSG_BUFS];   // buffer for sequenced parts of the log message
-    uint32_t      msg_len;                     // running message length
-    uint32_t      seq;                         // and running sequence count
-};
+
+//------------------------------------------------------------------------------
+// Tinsel Slot Allocations.
+// Slot 0 is used to send "normal" messages.
+// Slot 1 is used for supervisor messages (e.g. instrumentation)
+// Slot 2 is used for log messages.
+// Remaining slots are used for receive slots.
+//------------------------------------------------------------------------------
+#define P_MSG_SLOT      0
+#define P_SUPMSG_SLOT   1
+#define P_LOGMSG_SLOT   2
+#define P_RXSLOT_START  3
+//------------------------------------------------------------------------------
+
+const uint32_t p_logmsg_max_size = p_logmsg_pyld_size << P_LOG_MAX_LOGMSG_FRAG;
+const uint32_t p_logmsg_1msg_max_size = p_logmsg_pyld_size;
+const uint32_t p_logmsg_2msg_max_size = p_logmsg_pyld_size << 1;
+const uint32_t p_logmsg_3msg_max_size = p_logmsg_2msg_max_size + p_logmsg_pyld_size;
+
+void softswitch_trivial_log_handler(const char* &logStr);
 
 #ifdef TRIVIAL_LOG_HANDLER
-//inline void handler_log(int level, const char * msg, ...) {handler_log_msg message; if (!message.pack_fmtstr(msg)) message.send_msg();};
-template<typename... F> inline void handler_log(int level, const char * msg, F... args)  {handler_log_msg message; if (!message.pack_fmtstr(msg)) message.send_msg();};
+// Call a truly trivial log handler.
```

Comment:
Is the trivially trivial log handler called trivially, or truly trivially?

(Why don't we just call it a log handler and be done with it)

Reply:
It's trivial because it is format string only. The softswitch is designed to overload the trivial handler with something more featured as/when required, hence why the default is a "trivial" handler.